nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,232 @@
+
+        # Parameters for clean()
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+
+        # Parameters for joblib
+        self.memory = memory
+        self.memory_level = memory_level
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """Prepare signal extraction from regions.
+
+        All parameters are unused, they are for scikit-learn compatibility.
+        """
+            # This is not elegant but this is the easiest way to test it.
+        try:
+            for seed in self.seeds:
+                assert(len(seed) == 3)
+        except Exception as e:
+            if self.verbose > 0:
+                print('Seeds not valid, error' + str(e))
+            raise ValueError('Seeds must be a list of triplets of '
+                             'coordinates in native space.')
+        self.seeds_ = self.seeds
+        return self
+
+    def fit_transform(self, imgs, confounds=None):
+        return self.fit().transform(imgs, confounds=confounds)
+
+    def _check_fitted(self):
+        if not hasattr(self, "seeds_"):
+            raise ValueError('It seems that %s has not been fitted. '
+                             'You must call fit() before calling transform().'
+                             % self.__class__.__name__)
+
+    def transform(self, imgs, confounds=None):
+        """Extract signals from Nifti-like objects.
+
+        Parameters
+        ==========
+        imgs: Niimg-like object
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Images to process. It must boil down to a 4D image with scans
+            number as last dimension.
+
+        confounds: array-like, optional
+            This parameter is passed to signal.clean. Please see the related
+            documentation for details.
+            shape: (number of scans, number of confounds)
+
+        Returns
+        =======
+        signals: 2D numpy.ndarray
+            Signal for each region.
+            shape: (number of scans, number of regions)
+
+        """
+        self._check_fitted()
+
+        logger.log("loading images: %s" %
+                   _utils._repr_niimgs(imgs)[:200], verbose=self.verbose)
+        imgs = _utils.check_niimg(imgs)
+
+        if self.smoothing_fwhm is not None:
+            logger.log("smoothing images", verbose=self.verbose)
+            imgs = self._cache(image.smooth_img, func_memory_level=1)(
+                imgs, fwhm=self.smoothing_fwhm)
+
+        logger.log("extracting region signals", verbose=self.verbose)
+        signals = self._cache(
+            _signals_from_spheres, func_memory_level=1)(
+                self.seeds_, imgs, radius=self.radius, mask_img=self.mask_img)
+
+        logger.log("cleaning extracted signals", verbose=self.verbose)
+        signals = self._cache(signal.clean, func_memory_level=1
```

Comment:
Because there are 6 parameters and `clean` is equivalent to `id` if all of them are None.